src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,154 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
```

Comment:
Common mistakes says "Don’t iterate your nodes from the head to tail or vice versa"
So do you really need them in your class?

Reply:
Make your LinkedList to work as java.util.LinkedList (create Node objects to store data that have links to the left and to the right neighbour Node). Your implementation should have two pointers to the first and the last Node.           The task told to create this variables

Reply:
Don’t iterate your nodes from the head to tail or vice versa - its mean, that you need to choose from where will be a shorter path and iterate from tail or from head